Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1951,7 +1951,11 @@ extension BrowserViewController: LegacyTabDelegate {
             creditCardHelper.foundFieldValues = { fieldValues in
                 guard let tabWebView = tab.webView as? TabWebView else { return }
 
-                tabWebView.accessoryView?.reloadViewForCardAccessory()
+                // Delay so that this executes AFTER KeyboardHelperDelegate methods
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
```

Comment:
Could we lower it? and make it a constant instead of a magic number

Reply:
Do we not have a way of finding when keyboardhelper delegate gets called? Is there a method in that protocol we can use 🤔 

Just thinking out loud

Reply:
Lowered to 0.2 seconds, and have a property instead of a magic number now.

Reply:
I have to agree with Nish, there has to be another way to do this. We shouldn't rely on `DispatchQueue.main.asyncAfter` as much as possible since on different device this could create timing issues and what not.

Reply:
Posted investigation comments in the JIRA ticket, https://mozilla-hub.atlassian.net/browse/FXIOS-6297?focusedCommentId=678984

Reply:
I vote for option 1 😛 Sounds like the best approach so we don't rely on timing

Reply:
My vote's also for option 1 - @nbhasin2 do you think we can add an item for it in [the JS epic](https://mozilla-hub.atlassian.net/browse/FXIOS-5735).

And in the meantime, do we keep this delay?